File: 2021/10/08/rectangle_overlap.py
"""
https://leetcode.com/problems/rectangle-overlap/
Author: Andrew Jarombek
Date: 10/8/2021
"""

from typing import List


def is_rectangle_overlap(rec1: List[int], rec2: List[int]) -> bool:
    no_overlap = rec1[2] <= rec2[0] or rec1[0] >= rec2[2] or rec1[3] <= rec2[1] or rec1[1] >= rec2[3]
    return not no_overlap

File: 2022/05/02/rectangleOverlap.ts
/**
 * https://leetcode.com/problems/rectangle-overlap/
 * @author Andrew Jarombek
 * @since 5/2/2022
 */

function isRectangleOverlap(rec1: number[], rec2: number[]): boolean {
    if (rec1[0] == rec1[2] || rec2[0] == rec2[2] || rec1[1] == rec1[3] || rec2[1] == rec2[3])
        return false;

    let result = rec1[0] >= rec2[2] || rec1[3] <= rec2[1] || rec1[2] <= rec2[0] || rec1[1] >= rec2[3];
    return !result;
}